pkg/collector/corechecks/cluster/kubernetes_state.go

```diff
@@ -242,6 +300,18 @@ func (k *KSMCheck) Run() error {
 	return nil
 }
 
+// Stop releases the KSM Store resources (shutdown the KSM informers) and breaks the infinite Run loop
+// This is a long-running check, the scheduler guarantees calling this method when unscheduling the check
+func (k *KSMCheck) Stop() {
+	k.cancelStore()
+	k.stopCh <- struct{}{}
```

Comment:
Why not using `ctx.Done()` instead of maintaining your own channel? You'd only have to `cancelStore()` and `select` in previous file can use `ctx.Done()` to wait

Reply:
the only reason why it was done this way is just because I was solving the two problems separately (stoping the informers, and stoping the check itself) - I agree, sharing the same context makes more sense, thanks for pointing out 💯  changes made